nylas/handler/http_client.py

```diff
@@ -126,9 +130,9 @@ def _execute_download_request(
 
             # If we stream an iterator for streaming the content, otherwise return the entire byte array
             if stream:
-                return response
+                return _validate_download_response(response,stream)
 
-            return response.content if response.content else None
+            return _validate_download_response(response) if response.content else None
```

Comment:
@kraju3 do we need to make this change here? Looks like if it's `stream` we handle the response differently earlier in this function anyways.

Reply:
Hey @AaronDDM I was hoping to keep the same behavior plus return an error if the error code is different